funnel/models/commentset_membership.py

```diff
@@ -0,0 +1,110 @@
+from werkzeug.utils import cached_property
+
+from coaster.sqlalchemy import immutable, with_roles
+
+from . import User, db
+from .commentvote import Commentset
+from .helpers import reopen
+from .membership import ImmutableMembershipMixin
+
+__all__ = ['CommentsetMembership']
+
+
+class CommentsetMembership(ImmutableMembershipMixin, db.Model):
+    """Membership roles for users who are commentset users and subscribers."""
+
+    __tablename__ = 'commentset_membership'
+
```

Comment:
Required: `__data_columns__ = ('last_seen_at',)`

We're not using `.replace()` at the moment, but it will be required when multiple subscription states are added (such as a potential "mute replies"). The timestamp must be copied when the record is replaced.